src/components/main/Main.tsx

```diff
@@ -1,134 +1,42 @@
 import { positionContext } from 'context/positionContext';
-import { getCity } from 'services/getCity';
-import { getWeather } from 'services/getWeather';
-import { CurrentWeatherType, HourlyWeatherType, DailyWeatherType } from 'types/weatherTypes';
-import { LocationInfoType } from 'types/cityInfoType';
-import { defaultLocationInfo } from './defaultStates';
-import MainCard from './MainCard';
-import MainDailyCard from './MainDailyCard';
-import GraphHourly from './GraphHourly';
-import GraphDaily from './GraphDaily';
-import HourlySection from './HourlySection';
+
+import commonStyle from 'styles/commonStyles.css';
 
 const Main: FC = () => {
-  const [location, setLocation] = useState<LocationInfoType>(defaultLocationInfo);
-  const [currentWeather, setCurrentWeather] = useState<CurrentWeatherType | undefined>(undefined);
-  const [hourlyWeather, setHourlyWeather] = useState<HourlyWeatherType[]>([]);
-  const [dailyWeather, setDailyWeather] = useState<DailyWeatherType[]>([]);
-  const [selectDays, setSelectDays] = useState<string>('');
-  const [view, setView] = useState<'cards' | 'graph'>('cards');
+  const layer: layerState = useSelector(layerSelector);
 
   const { coordinates } = useParams();
   const [paramLongitude, paramLatitude] = (coordinates || '').split(',');
 
   const {
-    state: { position, positionError }
+    state: { position },
+    changePosition
   } = useContext(positionContext);
 
-  const resultLongitude = paramLongitude || position.longitude;
-  const resultLatitude = paramLatitude || position.latitude;
-
-  async function fetchData(lon: number | string, lat: number | string) {
-    Promise.all([
-      getCity(lon, lat),
-      getWeather<{ current: CurrentWeatherType }>('/current/', lon, lat),
-      getWeather<{ forecast: HourlyWeatherType[] }>('/forecast/hourly/', lon, lat, {
-        dataset: 'full'
-      })
-    ]).then(res => {
-      setLocation(res[0]);
-      setCurrentWeather(res[1].current);
-      setHourlyWeather(res[2].forecast);
-    });
-  }
-
-  const handleDaysSelected = (e: ChangeEvent<HTMLSelectElement>) => {
-    setSelectDays(e.target.value);
-    if (e.target.value) {
-      getWeather<{ forecast: DailyWeatherType[] }>(
-        '/forecast/daily/',
-        resultLongitude,
-        resultLatitude,
-        { periods: e.target.value, dataset: 'full' }
-      ).then(res => setDailyWeather(res.forecast));
-    }
-  };
+  const resultLongitude = +paramLongitude || position.longitude;
+  const resultLatitude = +paramLatitude || position.latitude;
 
   useEffect(() => {
-    if (resultLongitude && resultLongitude !== '0') {
-      fetchData(resultLongitude, resultLatitude);
+    if (resultLongitude !== position.longitude || resultLatitude !== position.latitude) {
+      changePosition(resultLatitude, resultLongitude);
     }
-  }, [resultLongitude, resultLatitude]);
+  }, [resultLongitude, position.longitude, resultLatitude, position.latitude]);
```

Comment:
1.  `resultLongitude/resultLatitude` variables need to be wrapped with `useMemo`, otherwise this effect will be called every render
2. if `resultLongitude/resultLatitude` variables not used anywhere except this effect, move its declaration inside effect callback. the same for `paramLongitude/paramLatitude `

Reply:
1. This component renders only when I open this page or when position is changed. In both cases component will render again. Of course for this two situations I need to know correct data about position. I think useMemo doesn't make sense in this case.
2. ok, I'll move this vars in useEffect callback. I think it'll solve this problem